tests/test_keyio.py

```diff
@@ -303,8 +303,8 @@ def test_keyjar_group_keys(self):
             keybundle_from_local_file(RSAKEY, "rsa", ["ver", "sig"]))
 
         verified_keys = ks.verify_keys("http://www.example.org")
-        assert len(verified_keys) == 4
-        assert len([k for k in verified_keys if k.kty == "oct"]) == 2
+        assert len(verified_keys) == 6
+        assert len([k for k in verified_keys if k.kty == "oct"]) == 4
```

Comment:
This seems to be wrong. The total number of keys in this `KeyJar` with `oct` type is 2. So this result contains some keys multiple times.

Reply:
Maybe. So there is the problem of method x_keys:
https://github.com/OpenIDC/pyoidc/blob/d0d502b7f479a3d721817e96f16278a4f1b9c477/src/oic/utils/keyio.py#L608-L613
line 611 - 4 keys, next will be 6

Reply:
In total you have received keys for all owners

Reply:
I am not sure what the reasoning for this is. Maybe a symmetric key lookup?
@rohe Can you please look into this?

Reply:
@rohe, can you please have a look? I think that this might be correct since we want to include our keys as well but I am not sure.

Reply:
When it comes to symmetric keys we definitely wants to include our own keys.
Looking at x_keys now I see at least one problem and that is as you say the problem of getting multiple copies of the same key. Since jwkest.jwk.Key instance has an __eq__ method we should be able to weed out those.
One reason why we get multiple copies is because if a key is marked to be for both signing and encrypting it is handled as 2 keys. And in the test the same key is marked to be both for signing and verification which is overkill since it's sufficient for a key to be mark for 'ver' to be used for both. Some cleaning up is in order.

Reply:
@ermakoves Can you please accommodate for the comment by @rohe so we can close this one?

Reply:
@tpazderka I'm not quite sure what I need to do. Share keys in the test for signed and encrypted? If so, then I do not think that this will clarify, and besides, why can not the key be able to do everything?

Reply:
@ermakoves Well I had another look at your tests (`test_get_inactive_ver`) and they pass for me even with the existing code. Could you please provide us with a test that is failing on the current master and shoulc pass according to you?

Reply:
run test on your master
>tests\test_keyio.py:354 (TestKeyJar.test_get_inactive_ver)
>self = <tests.test_keyio.TestKeyJar object at 0x051F3310>
>
>   def test_get_inactive_ver(self):
>        ks = KeyJar() 
>        ks['http://example.com'] = KeyBundle([{"kty": "oct", "key": "a1b2c3d4", "use": "sig"},
>                                              {"kty": "oct", "key": "a1b2c3d4", "use": "ver"}])    
>        ks['http://example.com'][0]._keys[1].inactive_since = 1   
>        key = ks.get_verify_key(owner='http://example.com')
>       assert len(key) == 2
>E       assert 1 == 2
>E        +  where 1 = len([<jwkest.jwk.SYMKey object at 0x051F3B10>])
>
>test_keyio.py:364: AssertionError'''

Reply:
![2018-05-14_18-07_d-git-forks-pyoidc_origin-pyo](https://user-images.githubusercontent.com/31726343/40005936-d72ccf76-57a1-11e8-8bad-ea9d508f0905.jpg)

Reply:
The keys are not duplicates in another test just returns all valid keys for all owners

Reply:
Yes, this is not a case of tests only, but a general principle. We can deal with it in different issue later on.